src/controllerimpl.h

```diff
@@ -46,8 +46,8 @@ class ControllerImpl : public QObject {
   // this method is called, the VPN client is in "connecting" state.  This
   // state terminates when the "connected" (or the "disconnected") signal is
   // received.
-  virtual void activate(const QList<Server>& serverList, const Device* device,
-                        const Keys* keys,
+  virtual void activate(const Server& server, const Device* device,
```

Comment:
NIT: Since we're gonna use every field of HopConnection, maybe we can simplify this to `activate(Hop,Device,Keys)`?

Reply:
I like this, a lot. Thanks for the suggestion.